src/libs/actions/Policy.js

```diff
@@ -144,40 +153,41 @@ function create(name = '', shouldAutomaticallyReroute = true) {
  *
  * This way, we ensure that there's no race condition between creating the new policy and fetching existing ones,
  * and we also don't have to wait for full policies to load before navigating to the new policy.
- *
- * @param {Boolean} [shouldCreateNewPolicy]
  */
-function getPolicyList(shouldCreateNewPolicy = false) {
-    let newPolicyID;
-    const createPolicyPromise = shouldCreateNewPolicy
-        ? create('', false)
-        : Promise.resolve();
-    createPolicyPromise
-        .then((policyID) => {
-            newPolicyID = policyID;
-            return API.GetPolicySummaryList();
-        })
+function getPolicyList() {
+    const policyCollection = {};
+    API.GetPolicySummaryList()
         .then((data) => {
             if (data.jsonCode === 200) {
-                const policyDataToStore = transformPolicyListToOnyxCollection(data.policySummaryList || []);
-                updateAllPolicies(policyDataToStore);
-            }
-
-            if (shouldCreateNewPolicy) {
-                Navigation.dismissModal();
-                Navigation.navigate(newPolicyID ? ROUTES.getWorkspaceInitialRoute(newPolicyID) : ROUTES.HOME);
+                lodashMerge(policyCollection, transformPolicyListToOnyxCollection(data.policySummaryList || []));
             }
 
             return API.GetPolicyList();
         })
         .then((data) => {
             if (data.jsonCode === 200) {
-                const policyDataToStore = transformPolicyListToOnyxCollection(data.policyList || []);
-                updateAllPolicies(policyDataToStore);
+                lodashMerge(policyCollection, transformPolicyListToOnyxCollection(data.policyList || []));
+            }
+        })
+        .finally(() => {
```

Comment:
Do we need a separate `finally` block here? Why not just move `updateAllPolicies(policyCollection)` into the if block of the previous `.then`?

Reply:
I don't know if I'd say "need", but it seems fine. The main difference is that if we hit a `catch()` by using `finally()` then whatever data we successfully managed to merged into `policyCollection` will be set regardless. If we did it in the previous `.then()` and the second API throws then no data will be set.